### PR TITLE
Docs: Fix Helm chart link path

### PR DESCRIPTION
**Pull request by saerosV**
**What this PR does / why we need it**:

Fix the link path leading to Loki's Helm chart, in `production/README.md`.

[Current broken link](https://github.com/grafana/loki/blob/main/production/production/helm/loki/README.md).

**Special notes for your reviewer**:

Hey @trevorwhitney, just found out the broken link added by #6985. 

**Checklist**
- [ ] Documentation added
- [ ] Tests updated
- [ ] Is this an important fix or new feature? Add an entry in the `CHANGELOG.md`.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

**Review by jeschkies — APPROVED** [review]

File: production/README.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ orchestrator from HashiCorp.
 ## Using Helm to deploy on Kubernetes
 
 Here are the Helm charts used to deploy Loki and Promtail to Kubernetes:
-- [Loki](./production/helm/loki/README.md) 
+- [Loki](../production/helm/loki/README.md#loki)
```
Comment (trevorwhitney) [inline production/README.md:56]:
```suggestion
- [Loki](./helm/loki/README.md#loki) 
```
Going up a level to go back into the same level is a bit confusing to me. Would this be more succinct?

**Review by trevorwhitney — APPROVED** [review]
LGTM, but I think there's a slightly more succinct way to express that path

File: production/README.md (hunk 1 above)
Comment (saerosV) [reply production/README.md]: It would, didn't notice the obvious choice at first haha. Thanks again.

**Comment by saerosV** [comment]
@trevorwhitney PR ready to be merged, suggestion applied. Thanks again!

**Pull request merged**